File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/email/handlers/AbstractMailboxChangesListener.java
package com.wearezeta.auto.common.email.handlers;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.Callable;

import com.wearezeta.auto.common.misc.Timedelta;
import java.util.logging.Logger;

import com.wearezeta.auto.common.log.ZetaLogger;

public abstract class AbstractMailboxChangesListener implements Callable<String> {
    protected Map<String, String> expectedHeaders = new HashMap<>();
    protected Object parentMBox;
    protected Timedelta timeout;
    protected Timedelta rejectMessagesBefore;

    protected Object getParentMbox() {
        return this.parentMBox;
    }

    protected final Logger log = ZetaLogger.getLog(this.getClass().getSimpleName());

    public AbstractMailboxChangesListener(Object parentMBox, Map<String, String> expectedHeaders, Timedelta timeout,
                                          Timedelta rejectMessagesBefore) {
        // clone map
        this.expectedHeaders.putAll(expectedHeaders);
        this.parentMBox = parentMBox;
        this.timeout = timeout;
        this.rejectMessagesBefore = rejectMessagesBefore;
    }

}

File: tests/testsCore/src/main/kotlin/com/wire/android/tests/core/email/AbstractMailboxChangesListener.kt
/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 */
package com.wire.android.tests.core.email

import com.wire.android.tests.core.config.BackendConnectionsReader
import com.wire.android.tests.core.utils.Timedelta
import com.wire.android.tests.core.utils.ZetaLogger
import java.util.concurrent.Callable
import java.util.logging.Logger

abstract class AbstractMailboxChangesListener(
    parentMBox: Any,
    expectedHeaders: Map<String, String>,
    protected val timeout: Timedelta,
    protected val rejectMessagesBefore: Timedelta
) : Callable<String> {

    protected val expectedHeaders: Map<String, String> = HashMap(expectedHeaders)
    protected val parentMBox: Any = parentMBox

    protected val log: Logger = ZetaLogger.getLog(AbstractMailboxChangesListener::class.simpleName)

    protected fun getParentMbox(): Any = this.parentMBox
}
